src/clover/mains_supply/__utils__.py

```diff
@@ -104,6 +105,20 @@ def get_intermittent_supply_status(  # pylint: disable=too-many-locals
                 else:
                     status.append(0)
         times = pd.DataFrame(status)
+
+        timesdaily = times % 24
+
+        zero_indices = timesdaily.index[timesdaily.iloc[:, 0] == 0]
+        n_values = grid_attributes.iloc[timesdaily.iloc[zero_indices, 0], 1]
+
+        print(zero_indices)
+        print(n_values)
+
+# Append additional rows with a value of 0 to df1
+
+        for i, n in zip(zero_indices, n_values):
```

Comment:
### ❓ Unclear what you're doing here
This is mostly due to variable names:
* `n` - is this... the number of... hours?
* `n_values` - not sure what this is as it's just a list of ones...?
  ```python
  (Pdb) n_values
  Name
  0    1
  0    1
  0    1
  0    1
  0    1
      ..
  0    1
  0    1
  0    1
  0    1
  0    1
  Name: std, Length: 51100, dtype: int64
  (Pdb) np.sum(n_values)
  51100
  (Pdb) len(n_values)
  51100
  ```
  
We should chat about what you're trying to do here as I'm not too sure...